src/drivers/MongoDriver.ts

```diff
@@ -1416,6 +1461,27 @@ export class MongoDriver implements DataStore {
           // match learning object by params.id
           $match: { _id: params.id },
         },
+        { $unwind: '$materials.files' },
```

Comment:
To avoid having to make a second query in the case were `materials.files` is an empty array or null, use the `preserveNullAndEmptyArrays` option for the `$unwind` operator.
https://docs.mongodb.com/manual/reference/operator/aggregation/unwind/

```suggestion
        { $unwind: {path: '$materials.files', preserveNullAndEmptyArrays: true} },
```